tools/bench-speedup.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/bash -e
+
+
+# cwd
+WD=`pwd`
+
+
+# move to rir home
+SCRIPTPATH=`cd $(dirname "$0") && pwd`
+BASE=`cd $SCRIPTPATH/.. && pwd`
+
+pushd $BASE/build
+
+
+# checks and setup
+if [[ $(git diff --shortstat 2> /dev/null | tail -n1) != "" ]]; then
+    echo "repo is dirty"
+    popd
+    exit 1
+fi
+
+if BRANCH=`git symbolic-ref --short -q HEAD`
+then
+    echo "on branch $BRANCH"
+else
+    echo "detached head"
+    popd
+    exit 1
+fi
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    USING_OSX=1
+fi
+
+RIR_REVISION=`git rev-parse HEAD`
+if [ "$#" == "1" ]; then
+    RIR_REVISION=$1
+fi
+PLAIN_DIR=$BASE/external/vanilla-r
+PLAIN_R=$PLAIN_DIR/bin/R
+MOD_DIR=$BASE/external/custom-r
+MOD_R=$MOD_DIR/bin/R
+BENCH=`ls ${BASE}/benchmarks/shootout/*/*.r`
+TIMEOUT=80
+RUNS=3
+OUT="$WD/benchmark-out"
+log="$OUT/benchmark_`git log $RIR_REVISION -n 1 --pretty=format:'%cd_%h' --date=format:'%Y-%m-%d_%H-%M-%S'`.csv"
+
+mkdir -p $OUT
+
+
+# checkout and build vanilla-r and custom-r, ensure packages not bytecompiled
+$BASE/tools/sync.sh --vanilla
+
+export R_NO_BASE_COMPILE=1
+
+pushd $PLAIN_DIR
+make clean
+if [ $USING_OSX -eq 1 ]; then
+  # Mac OSX
+    F77="gfortran -arch x86_64" FC="gfortran -arch x86_64" CXXFLAGS="-g3 -O2" CFLAGS="-g3 -O2" ./configure --enable-R-shlib --without-internal-tzcode --with-ICU=no --disable-byte-compiled-packages
+else
+    CXXFLAGS="-g3 -O2" CFLAGS="-g3 -O2" ./configure --with-ICU=no --disable-byte-compiled-packages
```

Comment:
Actually running RIR with R_ENABLE_JIT should recompile the package code. As soon rir is loaded then bytecompiled code and uncompiled code will be treated the same. So I think disabling precompiling packages is unfair to the stock R. That's also why we do one round of `execute()` without measuring as warmup. Otherwise it would be unfair to us, since we compile all the package code at runtime.